### PR TITLE
upgrade to angular 4.

### PR DESCRIPTION
**Pull request by aitboudad**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,19 +43,19 @@
...
-    "@angular/platform-server": "^2.3.1",
-    "@angular/router": "^3.3.1",
+    "@angular/common": "^4.0.0",
+    "@angular/compiler": "^4.0.0",
+    "@angular/compiler-cli": "^4.0.0",
+    "@angular/core": "^4.0.0",
+    "@angular/platform-browser": "^4.0.0",
+    "@angular/platform-browser-dynamic": "^4.0.0",
+    "@angular/platform-server": "^4.0.0",
+    "@angular/router": "^4.0.0",
```
Comment (marshall007) [inline package.json]: @aitboudad we'll need to use something along the lines of `>=4.0.0-beta <5.0.0` for the semver constraint here (which is what `@angular/cli` is using by default for projects generated with the `--ng4` flag).

`^4.0.0` does not capture betas or RCs.

Comment (aitboudad) [inline package.json]: The stable version will be released next week, IMO there is no reason to support betas or RCs.

Comment (mkoczorowski) [inline package.json]: hey, I'm a bit blocked atm. Can you make the changes available for download ? I would like to run some tests

Comment (mkoczorowski) [inline package.json]: I have just tested this with angular4 and angular 2 in both cases I got en error
RROR in Error encountered resolving symbol values statically. Calling function 'OpaqueToken', function calls are not supported. Consider replacing the function or lambda with a reference to an exported function, resolving symbol UIROUTER_ROOT_MODULE in /blah/node_modules/ui-router-ng2/lib/uiRouterNgModule.d.ts, resolving symbol _UIROUTER_INSTANCE_PROVIDERS

**Comment by wawyed** [comment]
Looking forward for this to be merged in ^^

File: package.json (hunk 1 above)
Comment (wawyed) [inline package.json]: Angular 4 is out. https://github.com/angular/angular/blob/master/CHANGELOG.md#400-invisible-makeover-2017-03-23 😄

Comment (amcdnl) [inline package.json]: Grabbed this and ran it, doesn't work with the current typescript version. Had to update to `2.2.1` for it to build.

**Comment by amcdnl** [comment]
FYI - This can be used now via `"ui-router-ng2": "ui-router/ng2#angular4-artifacts"`

**Comment by louwie17** [comment]
@amcdnl I still don't see all the reflected changes using: "ui-router-ng2": "github:ui-router/ng2#angular4-artifacts" as I still get a template parse warning, because UIView is still using <template> instead of <ng-template>.
Is there anyway I can get the other changes as well from npm?

**Comment by kolkov** [comment]
Updated to 4.0.1. This worked fine! 
@amcdnl, Thanks!

**Comment by wawyed** [comment]
What does it need to be done to be able to merge this in?

**Comment by amcdnl** [comment]
@christopherthielen - Can we make this master now?

**Comment by christopherthielen** [comment]
@aitboudad thanks for all your work on this.

**Pull request merged**